tasks/slave_maintenance/slave_operations/reconnect_slave.py

```diff
@@ -0,0 +1,106 @@
+from typing import Optional
+from tasks.task import Task
+from tasks.task_result import TaskResult
+from constants.jenkins import JENKINS_URLS
+from helper.sdk_helper.testdb_helper.slave_pool_helper import SlavePoolSDKHelper
+from helper.jenkins_helper.jenkins_helper_factory import JenkinsHelperFactory
+
+class ReconnectSlavesTask(Task):
+
+    def reconnect_slave(self, task_result: TaskResult, params: dict) -> None:
+        if "slave" not in params:
+            self.set_subtask_exception(ValueError("Invalid arguments passed"))
+        slave = params["slave"]
+        if "name" not in slave:
+            self.set_subtask_exception(ValueError(f"name key is missing for the slave {slave}"))
+        slave_name = slave["name"]
+
+        try:
+            slave_pool_helper = SlavePoolSDKHelper()
+            self.logger.info(f"Connection to Slave Pool successful")
+        except Exception as e:
+            exception = f"Cannot connect to Slave Pool using SDK : {e}"
+            self.set_subtask_exception(exception)
+
+        try:
+            slave_doc = eval(slave_pool_helper.get_slave_pool_doc(slave_name))
+            self.logger.info(f"Doc with slave name {slave_name} successfully found in slave-pool")
+        except Exception as e:
+            exception = f"Cannot find slave name {slave_name} doc in slave-pool : {e}"
+            self.set_subtask_exception(exception)
+
+        ipaddr = slave_doc["ipaddr"]
+        jenkins_host = slave_doc["jenkins_host"]
+
+        try:
+            jenkins_helper = JenkinsHelperFactory.fetch_helper(JENKINS_URLS[jenkins_host])
+        except Exception as e:
+            exception = f"Cannot fetch helper for slave {ipaddr}:{slave_name} : {e}"
+            self.set_subtask_exception(exception)
+
+        # Step 1 - Disconnect from jenkins
```

Comment:
Shouldn't this be reconnect to Jenkins or reconnect slave?

Reply:
Resolved in latest commit